pkg/backup/backup_pv_action.go

```diff
@@ -61,5 +62,19 @@ func (a *PVCAction) Execute(item runtime.Unstructured, backup *v1.Backup) (runti
 		GroupResource: kuberesource.PersistentVolumes,
 		Name:          pvc.Spec.VolumeName,
 	}
-	return item, []velero.ResourceIdentifier{pv}, nil
+	// remove dataSource if exists from prior restored CSI volumes
+	if pvc.Spec.DataSource != nil {
+		pvc.Spec.DataSource = nil
+	}
+	if pvc.Spec.DataSourceRef != nil {
+		pvc.Spec.DataSourceRef = nil
+	}
```

Comment:
Velero add label to items it restored
https://github.com/vmware-tanzu/velero/blob/65f99c12640414525476e02bb9302bf7c544a49d/pkg/restore/restore.go#L1853

Do we wanna limit this action to items we are certain are restored by velero?
Or perhaps limit to just those restored by velero-csi-plugin?
```suggestion
	if pvc.Spec.DataSource != nil {
		pvc.Spec.DataSource = nil
	}
	if pvc.Spec.DataSourceRef != nil {
		pvc.Spec.DataSourceRef = nil
	}
```

Reply:
I don't think prior restore is necessarily relevant. While the specific failure case we've seen involves prior restore, I think these fields would almost always result in restoring a PVC with stale or invalid content.

Reply:
Ok then retitle the PR to a broader scope, then suggest CSI restore as one of the failure case.